### DISCUSSION TITLE
Alternative dari df[!complete.cases(df),]? (R)

### DISCUSSION THREAD

**Discussion by AGelfP** [teamalgoritma/community#647] [Q&A] [answered]
Data saya mengenai ratio keuangan yang dimiliki oleh sebuah perusahaan yang ada di Indonesia dari akhir tahun 2011 hingga akhir tahun 2022.

Jadi saya ingin mencari perusahaan apa saja yang memiliki nilai NA, karena saya perlu mencari tahu apakah perusahaan yang tidak memiliki ratio keuangan pada periode tertentu telah mengalami kebangkrutan atau dimerger/diakuisisi oleh perusahaan lain. Biasanya, complete.cases sudah cukup karena menunjukkan row apa saja yang mengandung NA. Namun kali ini, karena data saya juga menunjukkan ratio keuangan sebuah perusahaan pada periode-periode berbeda, hasilnya menjadi seperti ini 

```
all_ratio_clean[!complete.cases(all_ratio_clean),]

  | Ticker<chr> | Company name<chr> |  
-- | -- | -- | --
82 | UNVR IJ Equity | Unilever Indonesia Tbk PT |  
83 | UNVR IJ Equity | Unilever Indonesia Tbk PT |  
84 | UNVR IJ Equity | Unilever Indonesia Tbk PT |  
85 | UNVR IJ Equity | Unilever Indonesia Tbk PT |  
86 | UNVR IJ Equity | Unilever Indonesia Tbk PT |  
87 | UNVR IJ Equity | Unilever Indonesia Tbk PT |  
88 | UNVR IJ Equity | Unilever Indonesia Tbk PT |  
89 | UNVR IJ Equity | Unilever Indonesia Tbk PT |  
90 | UNVR IJ Equity | Unilever Indonesia Tbk PT |  
315 | INTP IJ Equity | Indocement Tunggal Prakarsa Tbk PT
360 | BYAN IJ Equity | Bayan Resources Tbk PT |  
412 | BUMI IJ Equity | Bumi Resources Tbk PT |  
413 | BUMI IJ Equity | Bumi Resources Tbk PT |  
414 | BUMI IJ Equity | Bumi Resources Tbk PT |  
415 | BUMI IJ Equity | Bumi Resources Tbk PT |  
416 | BUMI IJ Equity | Bumi Resources Tbk PT |  
417 | BUMI IJ Equity | Bumi Resources Tbk PT |  
418 | BUMI IJ Equity | Bumi Resources Tbk PT |  
419 | BUMI IJ Equity | Bumi Resources Tbk PT |  
420 | BUMI IJ Equity | Bumi Resources Tbk PT

```

Apakah ada cara supaya outputnya itu hanya menunjukkan nama perusahaan sekali, seperti berikut?

```
UNVR IJ Equity | Unilever Indonesia Tbk PT
INTP IJ Equity | Indocement Tunggal Prakarsa Tbk PT
BYAN IJ Equity | Bayan Resources Tbk PT
BUMI IJ Equity | Bumi Resources Tbk PT
SIMP IJ Equity | Salim Ivomas Pratama Tbk PT
```

Terima kasih sebelumnya

**Comment by idoali** [comment, marked as answer]
Terima kasih atas pertanyaannya pak. 

Mungkin bisa dicoba dengan library dplyr pak. Fungsinya sendiri adalah group_by dan summarise. Berikut kurang lebih kodenya

```{r}
library(dplyr)

data %>% 
  filter(!complete.cases(.)) %>% 
  group_by(ticker, company_name) %>% 
  summarise(n = n()) %>% 
  select(-n)
```

Tentunya kode tersebut masih perlu bapak lakukan penyesuaian lagi dengan data bapak. Semoga dapat menjawab pak.
